tested/judge/core.py

```diff
@@ -402,3 +404,30 @@ def _process_results(bundle: Bundle, execution: Execution,
         if continue_ in (Status.TIME_LIMIT_EXCEEDED,
                          Status.MEMORY_LIMIT_EXCEEDED):
             return continue_
+
+
+def _copy_workdir_source_files(bundle: Bundle, common_dir: Path) -> List[str]:
+    """
+    Copy additional source files from the workdir to the common dir
+
+    :param bundle: Bundle information of the test plan
+    :param common_dir: The directory of the other files
+    """
+    prefix = bundle.lang_config.execution_prefix()
+    source_files = []
+
+    def recursive_copy(src: Path, dst: Path):
```

Comment:
Is there a reason we can't re-use (part of) this function: https://github.com/dodona-edu/universal-judge/blob/591a9a10587daf74c21e015bd850849d7d46d38d/tested/judge/execution.py#L176

Reply:
The new function only copies only source files which will propably be needed for the compilation.
The function referenced by you copies all the files in the workdir.

I will look if `shutil.copytree` can take a filefilter, but I'm no sure.